apps/trustlab/src/payload/collections/Users/hooks/protectRoleField.js

```diff
@@ -0,0 +1,64 @@
+import {
+  ROLE_ADMIN,
+  ROLE_AUTHOR,
+  ROLE_EDITOR,
+} from "@/trustlab/payload/access/roles";
+
+// Field hook that only runs on the role field
+export const protectRoleField = async ({
+  req,
+  data,
+  originalDoc,
+  operation,
+}) => {
+  // ensure first user is admin
+  if (operation === "create") {
+    const { docs } = await req.payload.find({
+      collection: "users",
+      limit: 1,
+    });
+    if (!docs.length) {
+      return ROLE_ADMIN;
+    }
+  }
```

Comment:
I think another option would be to customise the `create-first-user` [route](https://payloadcms.com/docs/admin/overview#admin-level-routes) to use a custom view that doesn't ask for role but set it to admin.

Reply:
@kilemensi This does not allow us to customise the `create-first-user` page, just the route. So if we use `/first-user`, the first user registration route would be `/admin/first-user`. Looking through their Discord, I came across this [message](https://discord.com/channels/967097582721572934/1084195567946702908/1087797496849367091) on customising the first user page:

> ... currently we don't have a component override in place for the create new user view. This would be a great addition if you want to open a feature request on our GitHub OR if you fancy building it, we would also happily accept a PR.
You can still use custom css on this page and possibly use a workaround.

Reply:
That is messed up then... but since we're probably the ones that will create the first user, no need to stress too much on this.